src/main/java/com/github/dockerjava/core/command/BuildImageCmdImpl.java

```diff
@@ -84,7 +84,11 @@ public BuildImageCmdImpl(BuildImageCmd.Exec exec, InputStream tarInputStream) {
 
     @Override
     public String getTag() {
-        return tag;
+        if (tags == null || tags.isEmpty()) {
+            return null;
+        }
+        // return first tag to be backward compatible
+        return tags.iterator().next();
```

Comment:
@KostyaSha ok, but then we end up with `withTag(String tag)` and new method `withTags(Set<String> tags)` - I don't think this is more elegant solution. But you decide.